### DISCUSSION TITLE
我想把整个ant design组件默认从蓝色换成绿色怎么换 所有组件 比如按钮颜色 input框鼠标悬浮颜色

### DISCUSSION THREAD

**Discussion by 2010363329** [ant-design/ant-design#46067] [Q&A] [answered]
目前只看到了设置单独组件方法：https://ant-design.antgroup.com/docs/react/customize-theme-cn
有没有设置全局所有组件的

**Comment by MadCcc** [comment, marked as answer]
这个就是全局的，保证组件都在这个 ConfigProvider 下就好了

**Reply by 2010363329** [reply]
我编写了组件 并且改了主题 我怎么保证所有组件都在这个ConfigProvider里面呢 小白刚接触我这个 能不能给个demo

**Comment by 2010363329** [comment]
```
const App: React.FC = () => (
  <ConfigProvider
    theme={{
      token: {
        // Seed Token，影响范围大
        colorPrimary: '#00000',
        borderRadius: 2,

        // 派生变量，影响范围小
        colorBgContainer: '#f6ffed',
      },
    }}
  >
    <Space>
      <Button type="primary">Primary</Button>
      <Button>Default</Button>
    </Space>
  </ConfigProvider>
);

export default App;
```
像这种代码 难道我每个组件都需要单独定制吗 没有通用修改方法吗

**Reply by afc163** [reply]
放到你应用的最外层容器上包裹一次就行了。

**Reply by 2010363329** [reply]
说实话 没怎么听懂（前端小白）具体怎么包裹呢 代码写在哪里呢 我是next.js项目搭建的 有没有具体demo可以看一下的

**Reply by 2010363329** [reply]
```
const App: React.FC = () => (
  <ConfigProvider
    theme={{
      token: {
        // Seed Token，影响范围大
        colorPrimary: '#000000',
        borderRadius: 2,

        // 派生变量，影响范围小
        colorBgContainer: '#f6ffed',
      },
    }}
  >
    <Space>
      <Button type="primary">Primary</Button>
      <Button>Default</Button>
    </Space>
  </ConfigProvider>
);

export default App;
```
比如我这段代码 如果不用这两个按钮 这里面的样式主题颜色已经应用到其他组件上了吗 还是不懂

**Reply by 2010363329** [reply]
我可以理解为放到app目录下的layout.jsx里面吗  然后全局渲染

**Comment by 2010363329** [comment]
https://ant-design.antgroup.com/docs/react/use-with-next-cn
参考这个会了 再见 感觉大家 祝大家生活愉快

**Reply by see7788** [reply]
> https://ant-design.antgroup.com/docs/react/use-with-next-cn 参考这个会了 再见 感觉大家 祝大家生活愉快

希望antd的主题全部是type="primary"，有没有办法不针对具体属性挨个设置？configProviderProps: ConfigProviderProps = {
    componentSize: "small",
    theme: {
      token: {
        colorInfoBorder: "#52c41a",
        colorBgContainer: '#52c41a',//组件的容器背景色，例如：默认按钮、输入框等。务必不要将其与 `colorBgElevated` 混淆。
        colorPrimary: '#52c41a',//品牌色
        colorPrimaryHover: "#52c41a",//	主色梯度下的悬浮态。
        colorIcon: '#52c41a',
        colorIconHover: '#52c41a',//	主色梯度下的悬浮态。
        colorPrimaryActive: '#52c41a',//	主色梯度下的深色激活态。
      },
      components: { }
    }
  }我设置如此多了，Radio.Group依然是蓝色的。我喜欢全部风格是绿色
